applications/skyepass.md

```diff
@@ -0,0 +1,251 @@
+
+### Product Details
+
+As a long-term password manager software user myself, I have been really frustrated of the services like LastPass, 1Password for either lack of functionalities or the idea of storing ones entire digital identity on their corporate servers. Existing open-source solutions are too technically complicated to use. 
+
+At the very basis of it, a password manager is no more than an encrypted database, an APP and a browser extension to interact with the database. 
+
+Therefore, our team create a new password manger software that has pretty and intuitive UI/UX, fully decentralized (i.e. our team own no backend servers) and hackable by providing an open API for people to develop extensions with. 
+
+Users who signup will first create a blockchain wallet and have the mnemonic (and a master password) as their sole identity credentials (pretty standard blockchain wallet stuff). Later, each database instance is called a `vault` (standard name for all password managers) and they are light-weight file based databases ([lowDB](https://github.com/typicode/lowdb) seems to be a great choice). User can be given options to choose the encryption behavior of their database. By default, the vault will be split into some pieces with a Shamir's secret sharing mechanism. 
+
+For instance, for a simplest sharing schema, when the vault is created to be shared with 2 other family members, the vault will be split into 4 parts (we call them `horcrux`, for those who do not know [the Harry Potter reference](http://harrypotter.shoutwiki.com/wiki/Horcrux#:~:text=A%20Horcrux%20is%20a%20powerful,one%20is%20to%20true%20immortality.) ) with a minimum quorum of 2 to be decrypted. One piece will be sent to IPFS without encryption, the other 3 pieces will be encrypted by each member's public key and be sent to IPFS. An NFT will be minted for the owner. The ID of the NFT will be the `vault ID` and the NFT's URI will be a metadata piece that only the owner can change as exampled below: 
+
+```json
+{
+    "pieces": 4,
+    "quorum": 2,
+    "nounce": 18,
+    "owner": "5Ef9ES1SLQZU4KucGsjvs8qexvppQFmDgHiqoqVptJ9nZDeu",
+    "members": [
+        "5EKj6S1SLQZU4KucGsjvs8qexvppQFmDgHiqsdsdtJ9nZ123",
+        "5EJK1S1SLQZjkLKucGsjvs8qekdjpQFmDgHiqoqVptJ9nZ978"
+    ],
+    "unencrypted_cid": "QmaTX2v2QuwkQvEERw17w2xACcr2WZhy9t3NAEPBjvqPSX",
+    "encrypted_cids": [
+        {
+            "cid":       "QmaTX2v2QuwkQvEERw17w2xACcr2WZhy9t3NAEPBjvqPSX",
+            "member": "5EKj6S1SLQZU4KucGsjvs8qexvppQFmDgHiqsdsdtJ9nZ123"
+        },{
+            "cid":       "QmaTX2v2QuwkQvEERw17w2xACcr2WZhy9t3NAEPBjvqPSX",
+            "member": "5EJK1S1SLQZjkLKucGsjvs8qekdjpQFmDgHiqoqVptJ9nZ978"
+        }, {
+            "cid":       "QmaTX2v2QuwkQvEERw17w2xACcr2WZhy9t3NAEPBjvqPSX",
+            "member": "5Ef9ES1SLQZU4KucGsjvs8qexvppQFmDgHiqoqVptJ9nZDeu"
+        }
+    ]
+}
+```
```

Comment:
Why NFTs? Are vault ownerships transferable? It sounds as if a metadata file hash would be sufficient.

Reply:
Kind of... so far, our smart contract implementation stores a metadata file hash, a `HashSet<(VaultId, AccountId)>`(to store those who can make changes to the URI) and a `HashMap<VaultId, AccountId>`(to store the owner of a vault). By NFTs, we are trying to borrow some common perceptions of typical NFTs: an NFT is an entry of a data structure that can be owned by a wallet; they are IDed by some unique IDs; a smart contract to access control different levels of privileges. 

As for transferability,  it's a decision we are trying to make, and, I'm eager to learn more of what people think. The technical difficulty is low. But should we do it? A tool to sell a high rank, full skins, "League of Legends" account, or share my premium Spotify account for a few dollars can be interesting, but we'd have to be careful to implementing some kinda Safe Harbor measurement to prevent us from potential violation of ToS from these companies.